pkg/apiserver/diagnose/diagnose.go

```diff
@@ -181,7 +181,13 @@ func (s *Service) reportHTMLHandler(c *gin.Context) {
 	}
 
 	// Serve report html directly from assets
-	d, err := s.uiAssetFS.Asset("build/diagnoseReport.html")
+	f, err := s.uiAssetFS.Open("diagnoseReport.html")
```

Comment:
maybe you can use [c.FileFromFS](https://github.com/gin-gonic/gin#serving-data-from-file)

Reply:
`c.FileFromFS` use `http.FileServer`, I just copy that code and replace it with `httpgzip.FileServer` 🤣 
BTW, why do we need to pass the assets file through to the service, not just use `uiserver#Handler` here?

Reply:
It it to serve the HTML file in a specific path ( so that `./data.js` will resolve to a correct location)